### ISSUE TITLE
Fix HEALTHCHECK syntax

### ISSUE THREAD

**Issue by thenewguy** [tox-dev/tox-docker#57]
I am experiencing a difficult time with healthchecks

According to https://docker-py.readthedocs.io/en/stable/api.html#docker.types.Healthcheck the syntax currently being used is for the "CMD" but we are using "CMD-SHELL".

For the following block:
```
[docker:elasticsearch:7.7.0]
healthcheck_cmd = curl http://localhost:9200/
healthcheck_interval = 3
healthcheck_timeout = 3
healthcheck_retries = 30
healthcheck_start_period = 5
```

We should end up with something like (set by hand):
```
"Healthcheck": {
	"Test": [
		"CMD-SHELL",
		"curl http://localhost:9200/",
	],
	...
},
```

But I actually observe:
```
"Healthcheck": {
	"Test": [
		"CMD-SHELL",
		"curl",
		"http://localhost:9200/"
	],
	...
},
```
which is the synatx for "CMD" and not "CMD-SHELL".

It causes the curl command to have invalid syntax:
```
"Health": {
	"Status": "unhealthy",
	"FailingStreak": 48,
	"Log": [
		{
			"Start": "2020-05-27T13:42:21.888274303Z",
			"End": "2020-05-27T13:42:23.012655345Z",
			"ExitCode": 2,
			"Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
		},
		{
			"Start": "2020-05-27T13:42:26.02263741Z",
			"End": "2020-05-27T13:42:26.558271891Z",
			"ExitCode": 2,
			"Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
		},
		{
			"Start": "2020-05-27T13:42:29.565061122Z",
			"End": "2020-05-27T13:42:30.312883216Z",
			"ExitCode": 2,
			"Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
		},
		{
			"Start": "2020-05-27T13:42:33.322260112Z",
			"End": "2020-05-27T13:42:34.318126196Z",
			"ExitCode": 2,
			"Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
		},
		{
			"Start": "2020-05-27T13:42:37.32493584Z",
			"End": "2020-05-27T13:42:38.447951637Z",
			"ExitCode": 2,
			"Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
		}
	]
}
```

The command is confirmed to work in the container via exec:

```

$ docker exec 4ed curl http://localhost:9200/
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0{
  "name" : "4edc756ea21c",
  "cluster_name" : "docker-cluster",
  "cluster_uuid" : "9Pmb2rI9RZaQADJMyozVew",
  "version" : {
    "number" : "7.7.0",
    "build_flavor" : "default",
    "build_type" : "docker",
    "build_hash" : "81a1e9eda8e6183f5237786246f6dced26a10eaf",
    "build_date" : "2020-05-12T02:01:37.602180Z",
    "build_snapshot" : false,
    "lucene_version" : "8.5.1",
    "minimum_wire_compatibility_version" : "6.8.0",
    "minimum_index_compatibility_version" : "6.0.0-beta1"
  },
  "tagline" : "You Know, for Search"
}
100   541  100   541    0     0  23912      0 --:--:-- --:--:-- --:--:-- 24590
​
```

Here is the inspect output:

```
$ docker inspect 4ed
[
    {
        "Id": "4edc756ea21c3a140491e951deafc7c87c4667b5c02b1d916b56fd0661b8f6e4",
        "Created": "2020-05-27T13:39:07.990104823Z",
        "Path": "/tini",
        "Args": [
            "--",
            "/usr/local/bin/docker-entrypoint.sh",
            "eswrapper"
        ],
        "State": {
            "Status": "running",
            "Running": true,
            "Paused": false,
            "Restarting": false,
            "OOMKilled": false,
            "Dead": false,
            "Pid": 24583,
            "ExitCode": 0,
            "Error": "",
            "StartedAt": "2020-05-27T13:39:10.571390402Z",
            "FinishedAt": "0001-01-01T00:00:00Z",
            "Health": {
                "Status": "unhealthy",
                "FailingStreak": 48,
                "Log": [
                    {
                        "Start": "2020-05-27T13:42:21.888274303Z",
                        "End": "2020-05-27T13:42:23.012655345Z",
                        "ExitCode": 2,
                        "Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
                    },
                    {
                        "Start": "2020-05-27T13:42:26.02263741Z",
                        "End": "2020-05-27T13:42:26.558271891Z",
                        "ExitCode": 2,
                        "Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
                    },
                    {
                        "Start": "2020-05-27T13:42:29.565061122Z",
                        "End": "2020-05-27T13:42:30.312883216Z",
                        "ExitCode": 2,
                        "Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
                    },
                    {
                        "Start": "2020-05-27T13:42:33.322260112Z",
                        "End": "2020-05-27T13:42:34.318126196Z",
                        "ExitCode": 2,
                        "Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
                    },
                    {
                        "Start": "2020-05-27T13:42:37.32493584Z",
                        "End": "2020-05-27T13:42:38.447951637Z",
                        "ExitCode": 2,
                        "Output": "curl: try 'curl --help' or 'curl --manual' for more information\n"
                    }
                ]
            }
        },
        "Image": "sha256:7ec4f35ab452cc25d01d9db6b89cbbd5a7ca37eb1f35017b50377874cd08a83e",
        "ResolvConfPath": "/var/lib/docker/containers/4edc756ea21c3a140491e951deafc7c87c4667b5c02b1d916b56fd0661b8f6e4/resolv.conf",
        "HostnamePath": "/var/lib/docker/containers/4edc756ea21c3a140491e951deafc7c87c4667b5c02b1d916b56fd0661b8f6e4/hostname",
        "HostsPath": "/var/lib/docker/containers/4edc756ea21c3a140491e951deafc7c87c4667b5c02b1d916b56fd0661b8f6e4/hosts",
        "LogPath": "/var/lib/docker/containers/4edc756ea21c3a140491e951deafc7c87c4667b5c02b1d916b56fd0661b8f6e4/4edc756ea21c3a140491e951deafc7c87c4667b5c02b1d916b56fd0661b8f6e4-json.log",
        "Name": "/unruffled_ishizaka",
        "RestartCount": 0,
        "Driver": "overlay2",
        "Platform": "linux",
        "MountLabel": "",
        "ProcessLabel": "",
        "AppArmorProfile": "docker-default",
        "ExecIDs": null,
        "HostConfig": {
            "Binds": null,
            "ContainerIDFile": "",
            "LogConfig": {
                "Type": "json-file",
                "Config": {}
            },
            "NetworkMode": "default",
            "PortBindings": null,
            "RestartPolicy": {
                "Name": "",
                "MaximumRetryCount": 0
            },
            "AutoRemove": false,
            "VolumeDriver": "",
            "VolumesFrom": null,
            "CapAdd": null,
            "CapDrop": null,
            "Capabilities": null,
            "Dns": null,
            "DnsOptions": null,
            "DnsSearch": null,
            "ExtraHosts": null,
            "GroupAdd": null,
            "IpcMode": "private",
            "Cgroup": "",
            "Links": null,
            "OomScoreAdj": 0,
            "PidMode": "",
            "Privileged": false,
            "PublishAllPorts": true,
            "ReadonlyRootfs": false,
            "SecurityOpt": null,
            "UTSMode": "",
            "UsernsMode": "",
            "ShmSize": 67108864,
            "Runtime": "runc",
            "ConsoleSize": [
                0,
                0
            ],
            "Isolation": "",
            "CpuShares": 0,
            "Memory": 0,
            "NanoCpus": 0,
            "CgroupParent": "",
            "BlkioWeight": 0,
            "BlkioWeightDevice": null,
            "BlkioDeviceReadBps": null,
            "BlkioDeviceWriteBps": null,
            "BlkioDeviceReadIOps": null,
            "BlkioDeviceWriteIOps": null,
            "CpuPeriod": 0,
            "CpuQuota": 0,
            "CpuRealtimePeriod": 0,
            "CpuRealtimeRuntime": 0,
            "CpusetCpus": "",
            "CpusetMems": "",
            "Devices": null,
            "DeviceCgroupRules": null,
            "DeviceRequests": null,
            "KernelMemory": 0,
            "KernelMemoryTCP": 0,
            "MemoryReservation": 0,
            "MemorySwap": 0,
            "MemorySwappiness": null,
            "OomKillDisable": false,
            "PidsLimit": null,
            "Ulimits": null,
            "CpuCount": 0,
            "CpuPercent": 0,
            "IOMaximumIOps": 0,
            "IOMaximumBandwidth": 0,
            "MaskedPaths": [
                "/proc/asound",
                "/proc/acpi",
                "/proc/kcore",
                "/proc/keys",
                "/proc/latency_stats",
                "/proc/timer_list",
                "/proc/timer_stats",
                "/proc/sched_debug",
                "/proc/scsi",
                "/sys/firmware"
            ],
            "ReadonlyPaths": [
                "/proc/bus",
                "/proc/fs",
                "/proc/irq",
                "/proc/sys",
                "/proc/sysrq-trigger"
            ]
        },
        "GraphDriver": {
            "Data": {
                "LowerDir": "/var/lib/docker/overlay2/362ddd168d6b538065c7c9eeb13412b12ae8744585fc0c5109d8ba0e5a63c961-init/diff:/var/lib/docker/overlay2/baf9df3e23683b4189c55e0160edff9effe29397193d330e105a4703216ff699/diff:/var/lib/docker/overlay2/c00ceab188cb2642a67d0abe5cdffa86435c5e18903766b4bdd76ae8c08fc669/diff:/var/lib/docker/overlay2/3d72b1bea24312ceeab8aa2be238960e51fec063a2cdd62a5991655b80bf5a94/diff:/var/lib/docker/overlay2/6cf1e29f3350d2e293b95d90e393dbeed247169a0dfc2133cd02362d94c8888c/diff:/var/lib/docker/overlay2/bbb793d12cc24b2a1c361dc5ae87f05815dd8459397ba39789e483f259e889a1/diff:/var/lib/docker/overlay2/12a04229be441eb3eb1789aa77567b836b021eb93883fc0dc8300d018d1c7dfa/diff:/var/lib/docker/overlay2/ee36d76ac96a52f721df43340d086412500392126a1b377ceef2508d37189fa5/diff:/var/lib/docker/overlay2/a47684d5768d97e528192f55983a23eb2ae05671daac88550d22f7aa3b3c5a27/diff:/var/lib/docker/overlay2/94b0e82e87b0dbbde2edb1bb19d839428455f34f80690c3552959a99aacd6e1c/diff",
                "MergedDir": "/var/lib/docker/overlay2/362ddd168d6b538065c7c9eeb13412b12ae8744585fc0c5109d8ba0e5a63c961/merged",
                "UpperDir": "/var/lib/docker/overlay2/362ddd168d6b538065c7c9eeb13412b12ae8744585fc0c5109d8ba0e5a63c961/diff",
                "WorkDir": "/var/lib/docker/overlay2/362ddd168d6b538065c7c9eeb13412b12ae8744585fc0c5109d8ba0e5a63c961/work"
            },
            "Name": "overlay2"
        },
        "Mounts": [],
        "Config": {
            "Hostname": "4edc756ea21c",
            "Domainname": "",
            "User": "",
            "AttachStdin": false,
            "AttachStdout": false,
            "AttachStderr": false,
            "ExposedPorts": {
                "9200/tcp": {},
                "9300/tcp": {}
            },
            "Tty": false,
            "OpenStdin": false,
            "StdinOnce": false,
            "Env": [
                "http_proxy=...snip...",
                "HTTP_PROXY=...snip...",
                "https_proxy=...snip...",
                "HTTPS_PROXY=...snip...",
                "POSTGRES_PASSWORD=pgpass",
                "POSTGRES_USER=pguser",
                "POSTGRES_DB=pgdb",
                "discovery.type=single-node",
                "ES_JAVA_OPTS=-Xms512m -Xmx512m",
                "apm-server.host=0.0.0.0:8200",
                "no_proxy=10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,localhost,127.0.0.1,elasticsearch",
                "NO_PROXY=10.0.0.0/8,172.16.0.0/12,192.168.0.0/16,localhost,127.0.0.1,elasticsearch",
                "PATH=/usr/share/elasticsearch/bin:/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
                "ELASTIC_CONTAINER=true"
            ],
            "Cmd": [
                "eswrapper"
            ],
            "Healthcheck": {
                "Test": [
                    "CMD-SHELL",
                    "curl",
                    "http://localhost:9200/"
                ],
                "Interval": 3000000000,
                "Timeout": 3000000000,
                "StartPeriod": 5000000000,
                "Retries": 30
            },
            "Image": "elasticsearch:7.7.0",
            "Volumes": null,
            "WorkingDir": "/usr/share/elasticsearch",
            "Entrypoint": [
                "/tini",
                "--",
                "/usr/local/bin/docker-entrypoint.sh"
            ],
            "OnBuild": null,
            "Labels": {
                "org.label-schema.build-date": "2020-05-12T02:01:37.602180Z",
                "org.label-schema.license": "Elastic-License",
                "org.label-schema.name": "Elasticsearch",
                "org.label-schema.schema-version": "1.0",
                "org.label-schema.url": "https://www.elastic.co/products/elasticsearch",
                "org.label-schema.usage": "https://www.elastic.co/guide/en/elasticsearch/reference/index.html",
                "org.label-schema.vcs-ref": "81a1e9eda8e6183f5237786246f6dced26a10eaf",
                "org.label-schema.vcs-url": "https://github.com/elastic/elasticsearch",
                "org.label-schema.vendor": "Elastic",
                "org.label-schema.version": "7.7.0",
                "org.opencontainers.image.created": "2020-05-12T02:01:37.602180Z",
                "org.opencontainers.image.documentation": "https://www.elastic.co/guide/en/elasticsearch/reference/index.html",
                "org.opencontainers.image.licenses": "Elastic-License",
                "org.opencontainers.image.revision": "81a1e9eda8e6183f5237786246f6dced26a10eaf",
                "org.opencontainers.image.source": "https://github.com/elastic/elasticsearch",
                "org.opencontainers.image.title": "Elasticsearch",
                "org.opencontainers.image.url": "https://www.elastic.co/products/elasticsearch",
                "org.opencontainers.image.vendor": "Elastic",
                "org.opencontainers.image.version": "7.7.0"
            }
        },
        "NetworkSettings": {
            "Bridge": "",
            "SandboxID": "bc96dad88a1f3792be1dd53715d610572ec6dfe1c6b53cd5fbd2b3c7b73bee98",
            "HairpinMode": false,
            "LinkLocalIPv6Address": "",
            "LinkLocalIPv6PrefixLen": 0,
            "Ports": {
                "9200/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "32816"
                    }
                ],
                "9300/tcp": [
                    {
                        "HostIp": "0.0.0.0",
                        "HostPort": "32815"
                    }
                ]
            },
            "SandboxKey": "/var/run/docker/netns/bc96dad88a1f",
            "SecondaryIPAddresses": null,
            "SecondaryIPv6Addresses": null,
            "EndpointID": "c34b6e1971296d96aeecae84ede7372e3660ee3fe55d4bff708337db9829be4d",
            "Gateway": "172.17.0.1",
            "GlobalIPv6Address": "",
            "GlobalIPv6PrefixLen": 0,
            "IPAddress": "172.17.0.4",
            "IPPrefixLen": 16,
            "IPv6Gateway": "",
            "MacAddress": "02:42:ac:11:00:04",
            "Networks": {
                "bridge": {
                    "IPAMConfig": null,
                    "Links": null,
                    "Aliases": null,
                    "NetworkID": "9c212bf694de0454368b2a50168c15014dfdef25c0f4d4ee8010786789c8ce24",
                    "EndpointID": "c34b6e1971296d96aeecae84ede7372e3660ee3fe55d4bff708337db9829be4d",
                    "Gateway": "172.17.0.1",
                    "IPAddress": "172.17.0.4",
                    "IPPrefixLen": 16,
                    "IPv6Gateway": "",
                    "GlobalIPv6Address": "",
                    "GlobalIPv6PrefixLen": 0,
                    "MacAddress": "02:42:ac:11:00:04",
                    "DriverOpts": null
                }
            }
        }
    }
]
```

**Comment by thenewguy** [comment]
Oddly [test_healthcheck.py](https://github.com/tox-dev/tox-docker/blob/master/test_healthcheck.py) seems like it would be catching this with [healthcheck_cmd = redis-cli ping | grep -q PONG
](https://github.com/tox-dev/tox-docker/blob/master/tox.ini#L44)

**Comment by dcrosta** [comment]
closed by 9cb22f651ef267551f350dc2259bd872a9df4fa9 in #73
